### DISCUSSION TITLE
Creating system tests for an ASP.net webservice's swagger.json file with OpenAPI.NET

### DISCUSSION THREAD

**Discussion by mcwarg** [microsoft/OpenAPI.NET#602] [Q&A]
Hello,
I have an application which has a REST api that I recently upgraded from .net core 3 to asp.net 5.0. 
The REST interface then moved from swagger 2.0 to openapi 3.0.1
I reached feature parity, except that I used to have a python project which used Bravado to run system tests on the older version. 
Bravado was basically grabbing the swagger.json of the webservice, and auto generating classes which I then relied on to run various call to my system, running a full system test on the actual webservice.
Bravado doesn't support openapi 3.0.1, so I am looking at alternative to reproduce this.
I am thinking of using OpenAPI.NET as a replacement. Setting up a separate project on my visual studio solution that would take my webservice frontend openapi file as input, generate classes on the fly from it that I could then use to generate REST calls to try out my system.

Is that something that OpenAPI.NET would be useful for? and if not, do you have alternative suggestions? Thanks (and hopefully this is not a wrong place to ask this^^).

**Comment by darrelmiller** [comment]
Sorry for missing this post.  Yes, we regularly use this library to translate OpenAPI V3 documents down to V2.  As long as you avoid features that are only supported in V3, you should have minimal issues.

**Reply by mcwarg** [reply]
Hello, sorry I don't think I understand your answer. I probably didn't explain what I wanted to achieve clearly.
I currently have a openapi 3.0.1 file which I am not trying to convert down to V2. 
What I am trying to find out is whether openAPI can consume that file at run time to provide for a dynamically generated client which I could use to run request and resolve responses for my application, so that I am able to keep running simple system tests against it.

I used to do this with this library: https://bravado.readthedocs.io/en/stable/ but that only supports V2, and since I migrated to V3, I am wondering whether OpenAPI.Net is a potential replacement.

**Reply by darrelmiller** [reply]
There is currently no support for creating HTTP requests and handling HTTP responses in this library.
